sunxspex/emission.py

```diff
@@ -0,0 +1,814 @@
+    numerical problems with discontinuities in the electron distribution function at eelow and
+    eebrk.
+
+    Parameters
+    ----------
+    model : `str`
+        Electron model either `thick-target` or `thin-target`
+    photon_energies : `numpy.array`
+        Array containing lower integration limits
+    maxfcn : `int`
+        Maximum number of points used in Gaussian quadrature integration
+    rerr : `float`
+        Desired relative error for integral evaluation
+    eelow : `float`
+        Low energy electron cutoff
+    eebrk : `float`
+        Break energy
+    eehigh : `float`
+        High energy electron cutoff
+    p : `float`
+        Slope below the break energy
+    q : `float`
+        Slope above the break energy
+    z : `float`
+        Mean atomic number of plasma
+    efd : `bool`
+        True - electron flux density distribution, False - electron density distribution. This
+        input is not used in the main routine, but is passed to Brm_Fthin()
+
+    Returns
+    -------
+    `tuple`
+        (DmlinO, irer) Array of integral evaluation and array of error flags
+
+    References
+    ----------
+    Initial version modified from SSW
+    `Brm2_DmlinO <https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_dmlino.pro>`_ and
+    `Brm2_Dmlin <https://hesperia.gsfc.nasa.gov/ssw/packages/xray/idl/brm2/brm2_dmlin.pro>`_.
+
+    """
+    mc2 = const.get_constant('mc2')
+    clight = const.get_constant('clight')
+
+    if not eelow <= eebrk <= eehigh:
+        raise ValueError(f'Condition eelow <= eebrek <= eehigh not satisfied '
+                         f'({eelow}<={eebrk}<={eehigh}).')
+
+    # Create arrays for integral sums and error flags.
+    intsum1 = np.zeros_like(photon_energies, dtype=np.float64)
+    ier1 = np.zeros_like(photon_energies, dtype=np.float64)
+    intsum2 = np.zeros_like(photon_energies, dtype=np.float64)
+    ier2 = np.zeros_like(photon_energies, dtype=np.float64)
+    intsum3 = np.zeros_like(photon_energies, dtype=np.float64)
+    ier3 = np.zeros_like(photon_energies, dtype=np.float64)
+
+    P1 = np.where(photon_energies < eelow)[0]
+    P2 = np.where(photon_energies < eebrk)[0]
+    P3 = np.where(photon_energies <= eehigh)[0]
+
+    # Part 1, below en_val[0] (usually eelow)
+    if model == 'thick-target':
+        if P1.size > 0:
+            print('Part1')
+            a_lg = np.log10(photon_energies[P1])
+            b_lg = np.log10(np.full_like(a_lg, eelow))
+            i = np.copy(P1)
+            intsum1, ier1 = integrate_part(model=model, maxfcn=maxfcn, rerr=rerr,
+                                           photon_energies=photon_energies,
+                                           eelow=eelow, eebrk=eebrk, eehigh=eehigh, p=p, q=q, z=z,
+                                           a_lg=a_lg, b_lg=b_lg, ll=i, efd=efd)
+
+            # ier = 1 indicates no convergence.
+            if sum(ier1) > 0:
```

Comment:
```suggestion
            if sum(ier1):
```
`> 0` is redundant here.